src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -111,18 +112,30 @@
     Text getDescription();
 
     /**
-     * Gets all subjects that have this permission set.
+     * Gets all loaded subjects that have this permission set.
      *
      * <p>If you want to know to which role-templates this permission is
      * assigned use {@link PermissionService#SUBJECTS_ROLE_TEMPLATE}.
      *
      * @param type The subject type identifier to use
-     * @return An immutable set of subjects that have this permission set
+     * @return An immutable map of subjects that have this permission set
      * @see SubjectCollection#getAllWithPermission(String)
      */
     Map<Subject, Boolean> getAssignedSubjects(String type);
 
     /**
+     * Searches for and returns all subjects that have this permission set.
+     *
+     * <p>If you want to know to which role-templates this permission is
+     * assigned use {@link PermissionService#SUBJECTS_ROLE_TEMPLATE}.
+     *
+     * @param type The subject type identifier to use
+     * @return An immutable map of subjects that have this permission set
+     * @see SubjectCollection#getAllWithPermission(String)
+     */
+    CompletableFuture<Map<SubjectReference, Boolean>> findAssignedSubjects(String type);
```

Comment:
Perhaps I'm mistaken here, but as far as I know, that "type" parameter simply refers to the id of the SubjectCollection to be queried.

The API has no real reference to "subject types", other than a few static collection names in the PermissionService class, which suggest standard identifiers for collections which store x type of subject.

I'm not sure if a catalog would be appropriate for these, as they are only default or "standard" names for those collections.
https://github.com/lucko/SpongeAPI/blob/feature/perms-futures/src/main/java/org/spongepowered/api/service/permission/PermissionService.java#L45-L85

It's quite normal that a permission service would hold more than just the standard collections.